src/proba_orientation.cpp

```diff
@@ -158,11 +159,50 @@ void propagate(bool latent, bool propagation, double I3,
 // param I3_list the 3-point mutual info (N * I'(X;Y;Z|{ui})) of each Triple
 // return vector<ProbaArray> Each ProbaArray is bound to a unshielded Triple
 vector<ProbaArray> getOriProbasList(const vector<Triple>& triples,
-    const vector<double>& I3_list, bool latent, bool degenerate,
+    const vector<double>& I3_list, int tau, 
+    const vector<miic::structure::Node>& nodes,
+    bool latent, bool latent_orientation, bool degenerate,
     bool propagation, bool half_v_structure) {
   int n_triples = triples.size();
   vector<ProbaArray> probas_list(n_triples);  // to be returned
   for (auto& p_array : probas_list) p_array.fill(0.5);
+  //
+  // In temporal mode, use time for orientation
+  //
+  if (tau >= 1) {
+    // When orientation can be deduced from time, The probability for the tail end 
+    // depends on latent variable discovery: 0 if not activated, 0.5 when activated
+    //
+    double proba_tail = 0;
+    if (latent || latent_orientation)
+      proba_tail = 0.5;
+    //
+    // Orient each triple using time when possible
+    //
+    std::regex lag_expr(".*_lag([0-9]+)$");
+    for (int i = 0; i < n_triples; i++) {
+      int nodeX_lag = stoi (std::regex_replace( nodes[ triples[i][0] ].name, lag_expr, "$1" ) );
+      int nodeZ_lag = stoi (std::regex_replace( nodes[ triples[i][1] ].name, lag_expr, "$1" ) );
+      int nodeY_lag = stoi (std::regex_replace( nodes[ triples[i][2] ].name, lag_expr, "$1" ) );
+      if (nodeX_lag < nodeZ_lag) {
+        probas_list[i][0] = 1;
+        probas_list[i][1] = proba_tail;
+      }
+      else if (nodeX_lag > nodeZ_lag) {
+        probas_list[i][0] = proba_tail;
+        probas_list[i][1] = 1;
+      }
+      
+      if (nodeZ_lag < nodeY_lag) {
+        probas_list[i][2] = 1;
+        probas_list[i][3] = proba_tail;
+      }
+      else if (nodeZ_lag > nodeY_lag) {
+        probas_list[i][2] = proba_tail;
+        probas_list[i][3] = 1;
+      }
+    }
+  }
```

Comment:
Here why do you still try to assign a probability for the lagged orientation? You've mentioned using "NA" for those probabilities in R, which is fine. So here imho it'd be better not to include them in the probability table at all, since they are irrelevant in this context. As suggest in the review comment above, you don't need to touch this file at all.